src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,35 @@
-import React from 'react';
+import React, { ChangeEvent, useCallback } from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { filterActions, FilterTodoStatus } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const { query } = useAppSelector(state => state.filter);
+  const dispatch = useAppDispatch();
+  const handleSelect = useCallback((event: ChangeEvent<HTMLSelectElement>) => {
+    dispatch(
+      filterActions.setTodoFilterStatus(event.target.value as FilterTodoStatus),
+    );
+  }, []);
+
+  const handleInput = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    dispatch(filterActions.setQuery(event.target.value));
+  }, []);
+
+  const handleClearQuery = useCallback(() => {
+    dispatch(filterActions.clearQuery());
+  }, []);
```

Comment:
Don't use `useCallback` without a reason. In this case, it even worsens the performance a little bit.

You can read more about when and when not to use it:
https://react.dev/reference/react/useCallback